### PR TITLE
PXP-515: [CLI] Develop the config secrets diff command

### PR DESCRIPTION
**Pull request by mfauzaan**
## Summary
This command will pull the latest development secrets file from Bunker, compare it with the local version, and display the diff if there are any changes made.

Screenshot 2023-04-06 at 7 35 38 PM

### Usage

To use the command, simply run: `wukong dev config diff`

Ticket: [PXP-515](https://mindvalley.atlassian.net/browse/PXP-515)

## What's Changed

- [x]  Implement `wukong dev config diff` command.
- [x] Add necessary utility functions to support the new command.
- [x] Moved common functions from config push to utils file

[PXP-515]: https://mindvalley.atlassian.net/browse/PXP-515?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/86#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by jk-gan — CHANGES_REQUESTED** [review]

**Review by jk-gan — CHANGES_REQUESTED** [review]
Need integration testing too. Other than that, LGTM! Good job! The `dev config diff` looks beautiful!

File: src/commands/dev/config/utils.rs
Comment (mfauzaan) [reply src/commands/dev/config/utils.rs:127]: oh hEyy, this one on bug PR: https://github.com/mindvalley/wukong-cli/pull/88

I'll send that one for review once this is merged. Didnt want to have so many context in one PR. Easier to review with smaller onces

**Comment by onimsha** [comment]
> Need integration testing too. Other than that, LGTM! Good job! The `dev config diff` looks beautiful!

I'll also wait for the integration tests before review the PR again.

**Review by onimsha — APPROVED** [review]

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/86#" title="Author self-approved">mfauzaan</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/86#pullrequestreview-1382636709" title="Approved">onimsha</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by onimsha** [comment]
LGTM 👍 .

@jk-gan please re-review this PR again.

File: Cargo.toml
Comment (jk-gan) [inline Cargo.toml:73]: The `serial_test` is only for development environment, so this should be installed es `dev-dependencies`

**Review by jk-gan — CHANGES_REQUESTED** [review]

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/86#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/86#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by jk-gan — APPROVED** [review]

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/86#pullrequestreview-1382737614" title="Approved">jk-gan</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/86#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/86#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/86#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by jk-gan — APPROVED** [review]
LGTM

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/86#pullrequestreview-1382769327" title="Approved">jk-gan</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/86#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
